File: problems/src/test/java/LongestCommonPrefixTest.java
import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class LongestCommonPrefixTest extends SampledTest {
    @Override
    public String getFileName() {
        return "longest_common_prefix.json";
    }

    @Test
    public void test() {
        for (TypedSample sample : getSamples(TypedSample[].class)) {
            for (LongestCommonPrefix solution : LongestCommonPrefix.values()) {
                assertWithMessage(sample.getMessage())
                    .that(solution.longestCommonPrefix(sample.input.strs))
                    .isEqualTo(sample.output);
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, String> {
        public static class Input {
            public String[] strs;
        }
    }
}

File: typescript/__tests__/problems1_100/longest-common-prefix.test.ts
import {strictEqual} from 'assert';
import SOLUTIONS from '../../src/problems1_100/longest-common-prefix';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    strs: string[],
  },
  output: string,
  message: string,
}

describe(
    'LongestCommonPrefix',
    () => {
      test(
          'test',
          () =>
              SOLUTIONS.forEach(solution => {
                getSamples().forEach((sample: Sample) => {
                  strictEqual(
                      solution.longestCommonPrefix(sample.input.strs),
                      sample.output,
                      sample.message,
                  );
                });
              }),
      );
    },
);
